doc/design/design_katib_xgboost.md

```diff
@@ -0,0 +1,31 @@
+# SQLFlow Trains Models via XGBoost in Katib
+
+## Requirement description:
+
+SQLFlow needs to provide users with a simple way to train their ML model. This function is supported by SQLFlow running XGBoost jobs on Katib.
```

Comment:
- What SQLFlow provides?
- How an existing SQLFlow sql looks like? Users need to specify model, hyper-parameters, input data, etc.
- This doc is to further reduce users' inputs by introducing hpo, to remove `with`.